webapp/app.py

```diff
@@ -166,3 +178,93 @@ def get_jira_tasks(webpage_id: int):
         return jsonify(tasks), 200
     else:
         return jsonify({"error": "Failed to fetch Jira tasks"}), 500
+
+
+@app.route("/api/remove_webpage", methods=["POST"])
+@validate()
+@login_required
+def remove_webpage(body: RemoveWebpageModel):
+    """
+    Remove a webpage based on its status.
+    This function handles the removal of a webpage from the system.
+    If the webpage is new and not in the codebase, it deletes the webpage and associated
+    reviewer records from the database. If the webpage pre-exists, it creates a Jira task to remove
+    the webpage from the code repository and updates the webpage status to "TO_DELETE".
+    Args:
+        body (RemoveWebpageModel): The model containing the details of the webpage to be removed.
+    Returns:
+        Response: A JSON response indicating the result of the operation.
+            - If the webpage is not found, returns a 404 error with a message.
+            - If the webpage is successfully deleted or a task is created, returns a 201 status with a success message.
+            - If there is an error during deletion, returns a 500 error with a message.
+    """
+    webpage_id = body.webpage_id
+
+    webpage = Webpage.query.filter(Webpage.id == webpage_id).one_or_none()
+    if webpage is None:
+        return jsonify({"error": "webpage not found"}), 404
+    if webpage.status == WebpageStatus.NEW:
+        try:
+            jira_tasks = JiraTask.query.filter_by(webpage_id=webpage_id).all()
+            if jira_tasks:
+                for task in jira_tasks:
+                    if app.config["JIRA"].change_issue_status(
+                        issue_id=task.jira_id,
+                        transition_id=JiraStatusTransitionCodes.REJECTED.value,
+                    ):
+                        JiraTask.query.filter_by(id=task.id).delete()
+
+                db.session.commit()
+            Reviewer.query.filter_by(webpage_id=webpage_id).delete()
+            Webpage.query.filter_by(id=webpage_id).delete()
+            db.session.commit()
+
+        except SQLAlchemyError as e:
+            # Rollback if there's any error
+            db.session.rollback()
+            print(f"Error deleting webpage: {str(e)}")  # log error
+            return jsonify({"error": f"unable to delete the webpage"}), 500
+
+        return (
+            jsonify(
+                {"message": f"request for removal of webpage is processed successfully"}
+            ),
+            201,
+        )
+
+    if webpage.status == WebpageStatus.AVAILABLE:
+        if not (
+            (
+                body.due_date
+                and datetime.strptime(body.due_date, "%Y-%m-%d") > datetime.now()
+            )
+            and (
+                body.reporter_id
+                and User.query.filter_by(id=body.reporter_id).one_or_none()
+            )
+        ):
```

Comment:
Pydantic already takes care of the fields validation. If these fields are not optional, we can update the types in the `RemoveWebPageModel` schema directly. In addition, we can also validate for correct timr values in the schema, as it will raise the error along with info related tot he specific field